api/runtime/sandbox/v1/sandbox.proto

```diff
@@ -62,6 +62,7 @@ message CreateSandboxRequest {
 	string bundle_path = 2;
 	repeated containerd.types.Mount rootfs = 3;
 	google.protobuf.Any options = 4;
+	string netns_path = 5;
```

Comment:
I added milestone 2.0 here because it looks unnecessary and we should think more on the additional fields.

IIUC, the netns_path should be in oci_runtime_spec which is in bundle or metadata extension....

Reply:
Yes, netns should be taken in either spec of metadata, but in the current implementation of CreateSandboxRequest:
1) "options" is used for PodSandboxConfig now which could not take the netns info.
2) We can make "options" into a "map[string]Any" so that it can store the metadata, but this field needs be changed and metadata is taken as a json type not a proto one when marshal and unmarshal.
3) There is no spec file under the bundle path at all.

Reply:
What about using rpc metadata to pass the nspath?

Both ttrpc-go and grpc-go packages supports the metadata in header.

* ttrpc-go https://pkg.go.dev/github.com/containerd/ttrpc#WithMetadata
    * We've used it for namespace. https://github.com/containerd/containerd/blob/v1.6.19/namespaces/ttrpc.go#L38
* grpc-go https://github.com/grpc/grpc-go/blob/master/Documentation/grpc-metadata.md#sending-metadata-1

I checked that ttrpc-rust and it's already supported metadata in header https://github.com/containerd/ttrpc-rust/blob/b2004f7e84ed52f3c13c24f101062e02069246ab/example/async-stream-server.rs#L26

All the non-finalized fields can be passthrough by metadata, like annotation.
For the netns_path, we can use `io.containerd.sandbox.netns_path` as key (for instance).
When it is ready, we can propose it as fixed field in the proto.

Reply:
@fuweid I'm not sure this is worth it. Sandbox API is experimental and subject to change without warranty up to 2.0 release, so we can add fields we believe reasonable. Given that CRI is currently takes care of networking, I see nothing wrong in passing this down to the underlying shim implementation. I'm not sure using ttrpc/grpc metadata headers is the right thing to do.